documentation/kubernetes-upgrade.md

```diff
@@ -0,0 +1,68 @@
+
+* On the master node, run the following
+
+        kubeadm upgrade plan
+  This checks that your cluster is upgradeable and fetches the versions available to upgrade to in an user-friendly way.
+
+* Now, pick a version to upgrade to and run.
+
+        kubeadm upgrade apply v1.9.2
+  You can refer the [link] (https://kubernetes.io/docs/tasks/administer-cluster/kubeadm-upgrade-1-9/) for detailed instructions.
+
+* Now that your control plane is upgraded, you need to upgrade your kubelets in turn.
+
+## Upgrading your master and node packages
+* For each host (referred to as $HOST below) in your cluster, upgrade kubelet by executing the following commands:
+
+        kubectl drain $HOST --ignore-daemonsets
+
+* Upgrade the Kubernetes package versions on the $HOST node by using a Linux distribution-specific package manager. You can refer this [link] for distro specific instructions.
+
+## Set up vSphere cloud provider to work for a Kubernetes cluster deployed in multi-VC environment during upgrade
+
+The upgrade instructions from Kubernetes 1.7/1.8 to 1.9 using kubeadm as described above works fine for a Kubernetes cluster deployed on a single vCenter.
+
+In 1.9, vSphere Cloud Provider has support for a single Kubernetes cluster deployed across multiple vCenters. So, if one needs to upgrade from Kubernetes cluster 1.7/1.8 deployed on single VC to Kubernetes cluster 1.9 deployed across multiple vCenters, few additiona steps need to be performed apart from those mentioned above.
+
+* Identify the nodes that needs to be moved to other vCenter.
+* Prepare the node for maintenance, marking it unschedulable and evicting the workload.
+
+        kubectl drain $NODE --ignore-daemonsets
+
+* When running this command against the master host, this error is expected and can be safely ignored (since there are static pods running on the master)
+
+        node "master" already cordoned
+        error: pods not managed by ReplicationController, ReplicaSet, Job, DaemonSet or StatefulSet (use --force to override): etcd-kubeadm, kube-apiserver-kubeadm, kube-controller-manager-kubeadm, kube-scheduler-kubeadm
+        
+* Migrate the node(s) from current vCenter to new vCenter. Both these vCenters should have atleast one shared datastore.
+* After successful migration, check the status of the kubelet on the node(s).
+
+        systemctl status kubelet
+
+* Bring the node(s) back online by marking it schedulable
+
+        kubectl uncordon $NODE
+
+* Update the vsphere.conf on master node to let vSphere Cloud Provider know about the nodes that are deployed in multiple VC's. Follow the instructions mentioned in the [link] (https://vmware.github.io/vsphere-storage-for-kubernetes/documentation/existing.html#step-5-create-the-vsphere-cloud-config-file-vsphereconf) to update the vsphere.conf file. This file is located at /etc/kubernetes directory.
```

Comment:
Shouldn't we have a step to say that the user has to remove vsphere.conf file from worker nodes and update the kubelet configuration flags?

Reply:
Its not mandatory right. Wouldn't VCP would ignore those fields.
As part of my testing, I have removed it. But I thought that it would work without these too. Do you see any problem?

Reply:
I have added the steps you mentioned.